src/test/ui/polymorphization/predicates.rs

```diff
@@ -18,7 +18,28 @@ where
     bar::<I>()
 }
 
+// In addition, check that `I` is considered used in `next::{{closure}}`, because `T` is used and
+// `T` is really just `I::Item`.
+
+pub(crate) struct Foo<'a, I, E>(I, &'a E);
+
+impl<'a, I, T: 'a, E> Iterator for Foo<'a, I, E>
+where
+    I: Iterator<Item = &'a (T, E)>,
```

Comment:
Can you add a test using `std::iter::Repeat<I>: Iter...` or something like that to check that we actually use `mark_ty` for `self_ty` here?

Reply:
I might be misunderstanding what you mean - but doesn't the existing `foo` function in this test handle that?

Reply:
Foo also has `ty::Param` as `self_ty`, doesn't it?

I want to test a type which contains a `ty::Param` as `self_ty` which is currently not tested afaict

Reply:
Fixed!